src/index.ts

```diff
@@ -162,5 +170,52 @@ export function transposeTarStream(tarStream: NodeJS.ReadableStream,
 
 		tarStream.pipe(extract)
 	})
+}
+
+/**
+ * getBuildThroughStream: Get a through stream, which when piped to will remove all
+ * extra output that is added as a result of this module transposing a Dockerfile.
+ *
+ * This function enables 'silent' emulated builds, with the only difference in output
+ * from a native build being that there is an extra COPY step, where the emulator is
+ * added to the container
+ */
+export function getBuildThroughStream(opts: TransposeOptions): NodeJS.ReadWriteStream {
+
+	// Regex to match against 'Step 1/5:', 'Step 1/5 :' 'Step 1:' 'Step 1 :'
+	// and all lower case versions.
+	const stepLineRegex = /^(?:step)\s\d(?:\/\d)?\s?:/i
+	const isStepLine = RegExp.prototype.test.bind(stepLineRegex)
 
+	// Function to strip the string matched with the regex above
+	const stripStepPrefix = (data: string): string => {
+		return data.substr(data.indexOf(':') + 1)
+	}
+
+	// Regex to match against the type of command, e.g. FROM, RUN, COPY
+	const stepCommandRegex = /^\s?(\w+)(:?\s)/i
+	const getStepCommand = RegExp.prototype.exec.bind(stepCommandRegex)
```

Comment:
This doesn't actually return the step command, it returns a regex result that you can get the step command from. Nicer to do the `[1].toUpperCase()` here imo, so the main logic later on is clearer and the step command parsing logic is all in one place.